File: Statistics/STSimDistributionValue.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports SyncroSim.StochasticTime

Public Class STSimDistributionValue
    Inherits DistributionValue

    Private m_StratumId As Nullable(Of Integer)
    Private m_SecondaryStratumId As Nullable(Of Integer)

    Public Sub New(
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal stratumId As Nullable(Of Integer),
        ByVal secondaryStratumId As Nullable(Of Integer),
        ByVal distributionTypeId As Integer,
        ByVal externalVariableTypeId As Nullable(Of Integer),
        ByVal externalVariableMin As Nullable(Of Double),
        ByVal externalVariableMax As Nullable(Of Double),
        ByVal value As Nullable(Of Double),
        ByVal valueDistributionTypeId As Nullable(Of Integer),
        ByVal valueDistributionFrequency As Nullable(Of DistributionFrequency),
        ByVal valueDistributionSD As Nullable(Of Double),
        ByVal valueDistributionMin As Nullable(Of Double),
        ByVal valueDistributionMax As Nullable(Of Double),
        ByVal valueDistributionRelativeFrequency As Nullable(Of Double))

        MyBase.New(
              iteration,
              timestep,
              distributionTypeId,
              externalVariableTypeId,
              externalVariableMin,
              externalVariableMax,
              value,
              valueDistributionTypeId,
              valueDistributionFrequency,
              valueDistributionSD,
              valueDistributionMin,
              valueDistributionMax,
              valueDistributionRelativeFrequency)

        Me.m_StratumId = stratumId
        Me.m_SecondaryStratumId = secondaryStratumId

    End Sub

    Public ReadOnly Property StratumId As Nullable(Of Integer)
        Get
            Return Me.m_StratumId
        End Get
    End Property

    Public ReadOnly Property SecondaryStratumId As Nullable(Of Integer)
        Get
            Return Me.m_SecondaryStratumId
        End Get
    End Property

End Class

File: src/Statistics/STSimDistributionValue.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2023 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using SyncroSim.StochasticTime;

namespace SyncroSim.STSim
{
    public class STSimDistributionValue : DistributionValue
    {
        private int? m_StratumId;
        private int? m_SecondaryStratumId;
        private int? m_TertiaryStratumId;

        public STSimDistributionValue(
            int? iteration, int? timestep, int? stratumId, int? secondaryStratumId, int? tertiaryStratumId, int distributionTypeId, 
            int? externalVariableTypeId, double? externalVariableMin, double? externalVariableMax, double? value, 
            int? valueDistributionTypeId, DistributionFrequency? valueDistributionFrequency, double? valueDistributionSD, 
            double? valueDistributionMin, double? valueDistributionMax, double? valueDistributionRelativeFrequency) : 
            base(iteration, timestep, distributionTypeId, externalVariableTypeId, externalVariableMin, 
                externalVariableMax, value, valueDistributionTypeId, valueDistributionFrequency, valueDistributionSD, 
                valueDistributionMin, valueDistributionMax, valueDistributionRelativeFrequency)
        {
            this.m_StratumId = stratumId;
            this.m_SecondaryStratumId = secondaryStratumId;
            this.m_TertiaryStratumId = tertiaryStratumId;
        }

        public int? StratumId
        {
            get
            {
                return this.m_StratumId;
            }
        }

        public int? SecondaryStratumId
        {
            get
            {
                return this.m_SecondaryStratumId;
            }
        }

        public int? TertiaryStratumId
        {
            get
            {
                return this.m_TertiaryStratumId;
            }
        }
    }
}
